app/src/pages/inside/testCaseLibraryPage/expandedOptions/createFolderModal/utils.ts

```diff
@@ -0,0 +1,47 @@
+import { Folder } from 'controllers/testCase/types';
+import { FolderWithFullPath } from './types';
+
+const addSubfoldersToFolder = (folder: Folder, otherPlainFolders: Folder[]) => {
+  const subFolders = otherPlainFolders.filter((f) => f.parentFolderId === folder.id);
+  folder.subFolders = subFolders.map((subFolder) =>
```

Comment:
Here you are mutating the value from store which is anti-pattern in redux (`addSubfoldersToFolder` is used in `buildFoldersMap`)

```
  const folders = useSelector(foldersSelector);

  ...

  const mappedFolders = useMemo(() => traverseFolders(buildFoldersMap(folders)), [folders]);
```

Reply:
I just getting value from redux and use modified version for building options, I didn't set it to anywhere just use modified version and pass to component that required proper version, initial value is the same, I guess it's not anti-pattern

Reply:
btw updated

Reply:
Once you mutate object it becomes anti-pattern. Values obtained from selectors should not be mutated this way - only via redux flow